ipyparallel/cluster/shellcmd.py

```diff
@@ -0,0 +1,860 @@
+                    env = env.strip('"')  # occurs under windows cmd
+                if "\\'" in env:
+                    env = env.replace("\\'", "'")  # replace quoted
+                env_dict = eval(env)
+                assert isinstance(env_dict, dict) is True
+                env = env_dict
+
+            assert isinstance(env, dict) is True  # make sure that env is a dictionary
+
+            # update environment
+            for key, value in env.items():
+                if value is not None and value != '':
+                    if isinstance(value, dict):
+                        os.environ[key] = json.dumps(value)
+                    else:
+                        os.environ[key] = str(value)
+                else:
+                    # unset entry if needed
+                    if key in os.environ:
+                        del os.environ[key]
+
+        if isinstance(start_cmd, str):
+            start_cmd = [start_cmd]
+
+        if self.platform == Platform.Windows:
+            # under windows we need to remove embracing double quotes
+            for idx, p in enumerate(start_cmd):
+                if not isinstance(p, str):
+                    start_cmd[idx] = str(p)
+                    continue
+                if p[0] == '"' and p[-1] == '"':
+                    start_cmd[idx] = p.strip('"')
+
+        self._log(f"start_cmd={start_cmd}  (use_break_way={self.use_break_way})")
+
+        if self.platform == Platform.Windows:
+            from subprocess import (
+                CREATE_BREAKAWAY_FROM_JOB,
+                CREATE_NEW_CONSOLE,
+                DEVNULL,
+                Popen,
+            )
+
+            flags = 0
+            if self.use_break_way != False:
+                flags |= CREATE_NEW_CONSOLE
+                flags |= CREATE_BREAKAWAY_FROM_JOB
+
+            pkwargs = {
+                'close_fds': True,  # close stdin/stdout/stderr on child
+                'creationflags': flags,
+            }
+            if output_file:
+                fo = open(output_file, "w")
+                pkwargs['stdout'] = fo
+                pkwargs['stderr'] = fo
+                pkwargs['stdin'] = DEVNULL
+
+            self._log(f"Popen(**pkwargs={pkwargs}")
+            p = Popen(start_cmd, **pkwargs)
+            self._log(f"pid={p.pid}")
+
+            print(f'__remote_pid={p.pid}__')
+            sys.stdout.flush()
+            if self.use_break_way == False:
+                self._log("before wait")
+                p.wait()
+                self._log("after wait")
+        else:
+            start_cmd = [self._linux_quote(x) for x in start_cmd]
+
+            if output_file:
+                nohup_start = f"nohup {' '.join(start_cmd)} >{output_file} 2>&1 </dev/null & echo __remote_pid=$!__"
+            else:
+                nohup_start = f"nohup {' '.join(start_cmd)} >/dev/null 2>&1 </dev/null & echo __remote_pid=$!__"
+            os.system(nohup_start)
```

Comment:
Very unlikley that we should call os.system anywhere. Popen objects should handle command-lines, etc. more reliably.